util/ipgen/lib.py

```diff
@@ -167,6 +170,31 @@ def __init__(self,
         self.param_values = IpConfig._check_param_values(
             template_params, param_values)
 
+    @staticmethod
+    def _check_object(obj: object, what: str) -> object:
+        """Check that obj is a Hjson-serializable object.
+
+        If not, raise a ValueError; the what argument names the object.
+
+        """
+        try:
+            # Round-trip objects through the JSON encoder to get the
+            # same representation no matter if we load the config from
+            # file, or directly pass it on to the template. Also, catch
+            # encoding/decoding errors when setting the object.
+            json = hjson.dumps(obj,
+                               ensure_ascii=False,
+                               use_decimal=True,
+                               for_json=True,
+                               encoding='UTF-8')
+            obj_checked = hjson.loads(json,
+                                      use_decimal=True,
+                                      encoding='UTF-8')
+        except TypeError as e:
+            raise ValueError('{} cannot be serialized as Hjson: {}'
+                             .format(what, str(e))) from None
```

Comment:
sorry i'm a bit late to this.  Assuming i'm understanding this correctly,  i think `_asdict()` may not completely work.  When going over this with @rswarbrick before, I distinctly remember certain class variables being excluded from the `asdict` output, especially those that were not originally part of hjson and were added later.